File: 20. Valid Parentheses/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        close = {")" : "(" , "}" : "{" , "]" : "[" }


        for value in s :

            if value in close:
                if stack and stack[-1] == close[value]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(value)

        return True if not stack else False
            

        

File: 20. Valid Parentheses/Valid Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
  const stack = [];
  const close = { ")": "(", "}": "{", "]": "[" };

  for (let value of s) {
    if (close.hasOwnProperty(value)) {
      if (stack.length && stack[stack.length - 1] === close[value]) {
        stack.pop();
      } else {
        return false;
      }
    } else {
      stack.push(value);
    }
  }

  return stack.length === 0;
};
